cpu/msp430-common/atomic.c

```diff
@@ -30,3 +31,21 @@ unsigned int atomic_set_return(unsigned int *val, unsigned int set)
     eINT();
     return old_val;
 }
+
+int atomic_inc(int *val)
+{
+    int old_val;
+    dINT();
```

Comment:
I did not check if the msp430 platform was up to date with regards to IRQ functions, since the above atomic_set_return also used eINT/dINT.

I will change this to match the Cortex-M0 implementation.